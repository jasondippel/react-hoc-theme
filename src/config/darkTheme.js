export const VERSION = 1
export const TYPE = 'dark'

export const VALUES = {
  /**********
   ** TEXT **
   **********/
  text: '#F5FAFC',
  textInverse: '#2F3638',
  textMeta: '#aeb4b5',
  textDisabled: '#999fa3',
  link: '#00aceb',

  /****************
   ** BACKGROUND **
   ****************/
  background100: '#2F3638',
  background200: '#2D3032',
  background300: '#222729',
  background400: '#384042',
  background500: '#3f4547',
  background600: '#FFFFFF', // NOTE: this is deprecated; will be removed in near future
  backgroundLight: '#FFFFFF',
  backgroundDark: '#1c2021',
  keyline: '#191d1f',

  /***********************
   ** ACCENTS - PRIMARY **
   ***********************/
  primary: '#007ba8',
  primaryAccent: '#074a64',
  primaryBackground: '#042e3e',

  /***********************
   ** ACCENTS - SUCCESS **
   ***********************/
  success: '#348113',
  successAccent: '#204f0c',
  successBackground: '#173a09',

  /***********************
   ** ACCENTS - WARNING **
   ***********************/
  warning: '#cc4100',
  warningAccent: '#7a2700',
  warningBackground: '#5c1d00',

  /*********************
   ** ACCENTS - ERROR **
   *********************/
  error: '#c20000',
  errorAccent: '#800000',
  errorBackground: '#5c0000',
}
