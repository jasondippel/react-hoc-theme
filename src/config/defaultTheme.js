export const VERSION = 'light.0'

// export const defaultTheme = {
//   /**********
//    ** TEXT **
//    **********/
//   text: '#2f3638',
//   textInverse: '#F5FAFC',
//   textMeta: '#5D6366',
//   textDisabled: '#70767a',
//   link: '#0078A4',
//
//   /****************
//    ** BACKGROUND **
//    ****************/
//   background100: '#FFFFFF',
//   background200: '#FAFAFA',
//   background300: '#F1F1F1',
//   background400: '#E3E7E8',
//   background500: '#D9DDDE',
//   background600: '#2f3638',
//   keyline: '#CDD2D4',
//
//   /***********************
//    ** ACCENTS - PRIMARY **
//    ***********************/
//   primary: '#00AEEF',
//   primaryAccent: '#B3E7FB',
//   primaryBackground: '#DEF4FD',
//
//   /***********************
//    ** ACCENTS - SUCCESS **
//    ***********************/
//   success: '#7AAF2C',
//   successAccent: '#C0DF95',
//   successBackground: '#EEF7E2',
//
//   /***********************
//    ** ACCENTS - WARNING **
//    ***********************/
//   warning: '#F4B21E',
//   warningAccent: '#FFE380',
//   warningBackground: '#FFF1C0',
//
//   /*********************
//    ** ACCENTS - ERROR **
//    *********************/
//   error: '#E60000',
//   errorAccent: '#F9BFBF',
//   errorBackground: '#FBE5E5',
// }

export const defaultTheme = {
  /**********
   ** TEXT **
   **********/
  text: '#F5FAFC',
  textInverse: '#2F3638',
  textMeta: '#aeb4b5',
  textDisabled: '#81888c',
  link: '#008cbf',

  /****************
   ** BACKGROUND **
   ****************/
  background100: '#2F3638',
  background200: '#2D3032',
  background300: '#222729',
  background400: '#384042',
  background500: '#3f4547',
  background600: '#FFFFFF',
  keyline: '#191d1f',

  /***********************
   ** ACCENTS - PRIMARY **
   ***********************/
  primary: '#008abd', // '#00AEEF' // 37%
  primaryAccent: '#064760', // '#B3E7FB' // 20%
  primaryBackground: '#042e3e', // '#DEF4FD' // 13%

  /***********************
   ** ACCENTS - SUCCESS **
   ***********************/
  success: '#3b9216',
  successAccent: '#204f0c',
  successBackground: '#173a09',

  /***********************
   ** ACCENTS - WARNING **
   ***********************/
  warning: '#c76a00',
  warningAccent: '#5c3100',
  warningBackground: '#3d2100',

  /*********************
   ** ACCENTS - ERROR **
   *********************/
  error: '#e00000',
  errorAccent: '#5c0000',
  errorBackground: '#3d0000',
}
