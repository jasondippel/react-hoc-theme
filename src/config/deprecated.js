export const DEPRECATED_FIELDS = {
  // Example of structure
  // [keyPath]: {
  //   upgradeInstructions: 'Use "X/Y/Z" instead',
  // },
}
