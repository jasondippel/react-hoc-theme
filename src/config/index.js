import {
  VALUES as LIGHT_THEME_VALUES,
  VERSION as LIGHT_THEME_VERSION,
  TYPE as LIGHT_THEME_TYPE,
} from './lightTheme'
import {
  VALUES as DARK_THEME_VALUES,
  VERSION as DARK_THEME_VERSION,
  TYPE as DARK_THEME_TYPE,
} from './darkTheme'

const THEME_UPDATE_EVENT = 'THEME_UPDATE_EVENT'

const LIGHT_THEME = {
  type: LIGHT_THEME_TYPE,
  version: LIGHT_THEME_VERSION,
  values: LIGHT_THEME_VALUES,
}

const DARK_THEME = {
  type: DARK_THEME_TYPE,
  version: DARK_THEME_VERSION,
  values: DARK_THEME_VALUES,
}

const DEFAULT_THEME = LIGHT_THEME
const DEFAULT_THEME_VALUES = LIGHT_THEME_VALUES
const DEFAULT_THEME_VERSION = LIGHT_THEME_VERSION
const DEFAULT_THEME_TYPE = 'default'

export {
  THEME_UPDATE_EVENT,
  // default theme
  DEFAULT_THEME,
  DEFAULT_THEME_VALUES,
  DEFAULT_THEME_TYPE,
  DEFAULT_THEME_VERSION,
  // light theme
  LIGHT_THEME,
  LIGHT_THEME_VALUES,
  LIGHT_THEME_TYPE,
  LIGHT_THEME_VERSION,
  // dark theme
  DARK_THEME,
  DARK_THEME_VALUES,
  DARK_THEME_TYPE,
  DARK_THEME_VERSION,
}
