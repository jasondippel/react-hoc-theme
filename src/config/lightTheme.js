export const VERSION = 1
export const TYPE = 'light'

export const VALUES = {
  /**********
   ** TEXT **
   **********/
  text: '#181A1B',
  textInverse: '#F5FAFC',
  textMeta: '#4e5356',
  textDisabled: '#61696b',
  link: '#007099',

  /****************
   ** BACKGROUND **
   ****************/
  background100: '#FFFFFF',
  background200: '#FAFAFA',
  background300: '#F1F1F1',
  background400: '#E3E7E8',
  background500: '#D9DDDE',
  backgroundLight: '#FFFFFF',
  backgroundDark: '#181A1B',
  keyline: '#CDD2D4',

  /***********************
   ** ACCENTS - PRIMARY **
   ***********************/
  primary: '#00AEEF',
  primaryAccent: '#B3E7FB',
  primaryBackground: '#DEF4FD',

  /***********************
   ** ACCENTS - SUCCESS **
   ***********************/
  success: '#7AAF2C',
  successAccent: '#C0DF95',
  successBackground: '#EEF7E2',

  /***********************
   ** ACCENTS - WARNING **
   ***********************/
  warning: '#f4b71e',
  warningAccent: '#FFE380',
  warningBackground: '#FFF1C0',

  /*********************
   ** ACCENTS - ERROR **
   *********************/
  error: '#FF1A1A',
  errorAccent: '#F9BFBF',
  errorBackground: '#FBE5E5',
}
