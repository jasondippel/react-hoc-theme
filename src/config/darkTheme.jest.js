import { VERSION, TYPE, VALUES } from './darkTheme'

describe('darkTheme', () => {
  test('version is unchanged', () => expect(VERSION).toEqual(1))

  test('type is dark', () => expect(TYPE).toEqual('dark'))

  test('values remain unchanged', () => expect(VALUES).toMatchSnapshot(VALUES))
})
