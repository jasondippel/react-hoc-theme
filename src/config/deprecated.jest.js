import { DEPRECATED_FIELDS } from './deprecated'

describe('DEPRECATED_FIELDS', () => {
  test('all deprecated fields have upgrade instructions', () => {
    Object.keys(DEPRECATED_FIELDS).map(k => {
      expect(DEPRECATED_FIELDS[k].upgradeInstructions).toBeDefined()
    })
  })
})
