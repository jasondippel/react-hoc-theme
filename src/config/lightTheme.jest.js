import { VERSION, TYPE, VALUES } from './lightTheme'

describe('lightTheme', () => {
  test('version is unchanged', () => expect(VERSION).toEqual(1))

  test('type is light', () => expect(TYPE).toEqual('light'))

  test('values remain unchanged', () => expect(VALUES).toMatchSnapshot(VALUES))
})
