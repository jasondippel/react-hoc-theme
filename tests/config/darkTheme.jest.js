import { VERSION, TYPE, VALUES } from '../../src/config/darkTheme'

describe('darkTheme', () => {
  test('version is unchanged', () => expect(VERSION).toEqual(1))

  test('type is dark', () => expect(TYPE).toEqual('dark'))

  test('values remain unchanged', () => expect(VALUES).toMatchSnapshot(VALUES))
})
