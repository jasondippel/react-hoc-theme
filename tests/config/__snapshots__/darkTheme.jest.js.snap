// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`darkTheme values remain unchanged 1`] = `
Object {
  "colors": Object {
    "background100": "#2F3638",
    "background200": "#2D3032",
    "background300": "#222729",
    "background400": "#384042",
    "background500": "#3f4547",
    "backgroundDark": "#181A1B",
    "backgroundLight": "#FFFFFF",
    "error": "#c20000",
    "errorAccent": "#800000",
    "errorBackground": "#5c0000",
    "keyline": "#191d1f",
    "link": "#00aceb",
    "primary": "#007ba8",
    "primaryAccent": "#074a64",
    "primaryBackground": "#042e3e",
    "success": "#348113",
    "successAccent": "#204f0c",
    "successBackground": "#173a09",
    "text": "#F5FAFC",
    "textDisabled": "#999fa3",
    "textInverse": "#181A1B",
    "textMeta": "#aeb4b5",
    "warning": "#cc4100",
    "warningAccent": "#7a2700",
    "warningBackground": "#5c1d00",
  },
}
`;
